kubetest/util.go

```diff
@@ -168,6 +170,26 @@ func finishRunning(cmd *exec.Cmd) error {
 	}
 }
 
+/*
+cmd: command to retry
+timeout: how long to retry for
+*/
+func retryFinishRunning(cmd *exec.Cmd, timeout time.Duration) error {
+	stepName := strings.Join(cmd.Args, " ")
+	var err error
+
+	for start := time.Now(); time.Since(start) < timeout; {
+		thisCmd := *cmd
+		err = finishRunning(&thisCmd)
```

Comment:
The timeout is passed from the command line args [here](https://github.com/kubernetes/test-infra/pull/2576/files#diff-d1c748936b92e3673a2a848a2773a08eR161)